ast.cpp

```diff
@@ -28,12 +32,24 @@ class ProgramNode : public AstNode {
   ProgramNode(std::vector<std::unique_ptr<ExprNode>>&& exprs)
       : exprs_{std::move(exprs)} {}
 
+  void CodeGen() {
+    /* qbe main */
+    output << "export function w $main() {" << std::endl;
+    output << "@start" << std::endl;
+  }
```

Comment:
```suggestion
  void CodeGen() {
    /* qbe main */
    output << "export function w $main() {" << std::endl;
    output << "@start" << std::endl;
    for (const auto& expr : exprs_) {
      expr->CodeGen();
    }
    output << " ret 0" << std::endl;
    output << "}";
  }

```
`CodeGen` should recursively call `CodeGen` on its sub-nodes, just like what `Dump` does.
`MainEnd` can then be inlined into `CodeGen`.